src/models/task/impls.rs

```diff
@@ -82,6 +88,47 @@ impl Ord for Task {
 }
 
 impl Task {
+    /// Create new task
+    /// ## Parmeters:
+    /// - task_id: id of the task
+    /// - title: title of the task
+    /// - duration: duration of the task
+    /// - goal: goal of the task
+    /// - timeline: timeline of the task
+    /// - status: status of the task
+    /// - timeframe: Start and deadline of a task
```

Comment:
This way of documenting is used in the [Rust docs](https://doc.rust-lang.org/rust-by-example/meta/doc.html) - so I understand why we would try this. However - this documentation style is not used in the standard library.  

I think I err on the side of the standard library. **A function doc comment should be restricted to a simple phrase in the general description, and possibly a thing or two that is important to pay attention to.**  
Arguments:  
- Clean code guidelines state the number of params should be kept small - so a short description is easy.  
There is no need for a long list of params that need documentation.  
- Documenting the parameters is currently not auto-checked and thus leads to out-of-date docs.  
- If the params are structs, the docs are in the struct. Providing a summary would lead to duplication.

Reply:
I fixed that  by encapsulating the many parameters into single struct `NewTask`. Check and give me a feedback

Reply:
Well, the doc comments are gone now so yeah!  
The `NewTask` was not necessary in my opinion - it doesn't really add anything except indirection - but I guess it's temporary anyway so good.